src/apis/term.cpp

```diff
@@ -449,6 +449,45 @@ static int term_getPixels(lua_State* L) {
     return 1;
 }
 
+static int term_fillPixels(lua_State* L) {
+    lastCFunction = __func__;
+
+    Computer* computer = get_comp(L);
+    Terminal* term = computer->term;
+
+    const int init_x = (int) luaL_checkinteger(L, 1),
+              init_y = (int) luaL_checkinteger(L, 2),
+              end_w = (int) luaL_checkinteger(L, 3),
+              end_h = (int) luaL_checkinteger(L, 4),
+              color = (int) luaL_checkinteger(L, 5);
+
+    if (end_w < 0) return luaL_argerror(L, 3, "width must be positive");
+    else if (end_h < 0) return luaL_argerror(L, 4, "height must be positive");
+    else if (color < 0) return 0;
+    else if (term->mode == 2 ? color > 255 : log2i(color) > 15)
+        return luaL_argerror(L, 5, "color index out of bounds");
+
+    std::lock_guard<std::mutex> lock(term->locked);
+
+    const int index = term->mode == 2 ? color : log2i(color);
+
+    for (int h = 0; h < end_h; h++) {
+        const int y = init_y + h;
+
+        if (y < 0 || y >= term->height * Terminal::fontHeight) continue;
+
+        for (int w = 0; w < end_w; w++) {
```

Comment:
Using `memset` here would allow the loop to be optimized by the C library.

```c++
memset(&term->pixels[y][init_x], index, end_w);
```

A bit more than that would be necessary to respect sizes; check how I did it in [`term_drawPixels`](https://github.com/MCJack123/craftos2/blob/master/src/apis/term.cpp#L387-L389).

Reply:
You did it wrong anyway